### PR TITLE
feat(carousel): allow control mouse event

### PR DESCRIPTION
**Pull request by DheerajVislavath**
This commit resolves issue #1163 , which allows the user to enable or disable the mouse events. Please review the pull requests and let me know if everything is good to be merged into. Thanks

### CODE REVIEW

**Comment by DheerajVislavath** [comment]
Please review  code and merge the changes. This will be an important feature.

**Comment by DheerajVislavath** [comment]
@benouat Thank you for your suggestions.
But , Carousel already has pause() and cycle() methods to set the interval values to 0 or give default interval.  Why not make use of those methods ?

**Comment by benouat** [comment]
> Carousel already has pause() and cycle() methods to set the interval values to 0 or given default interval. Why not make use of those methods ?

You're right !! I did not see these ones.

Then it should be even easier. You just need to introduce the `pauseOnHover` as an input &amp; config

**Comment by DheerajVislavath** [comment]
I implemented your suggested change. Please review and process to merge the code changes. 
Thank you.

File: src/carousel/carousel.spec.ts [hunk 1]
```diff
@@ -321,9 +322,53 @@ describe('ngb-carousel', () => {
...
+       tick(6000);
+       fixture.detectChanges();
+       expectActiveSlides(fixture.nativeElement, [true, false]);
+       discardPeriodicTasks();
+     }));
+
   it('should pause / resume slide change with time passage on mouse enter / leave', fakeAsync(() => {
        const html = `
-      <ngb-carousel>
+      <ngb-carousel [pauseOnHover]='pauseOnHover'>
```
Comment (benouat) [inline src/carousel/carousel.spec.ts]: I don't see the point about adding that here. It basically duplicate with your added test.
Both `'should listen to mouse events based on pauseOnHover attribute'` and `'should pause / resume slide change with time passage on mouse enter / leave'` are the same.

File: src/carousel/carousel.ts [hunk 2]
```diff
@@ -39,8 +39,8 @@ export class NgbSlide {
     'class': 'carousel slide',
     '[style.display]': '"block"',
     'tabIndex': '0',
-    '(mouseenter)': 'pause()',
-    '(mouseleave)': 'cycle()',
+    '(mouseenter)': 'mouseEventPause()',
```
Comment (benouat) [inline src/carousel/carousel.ts]: I would rename it to `onMouseEnter()` as for now it won't be related to pause all the time.

File: src/carousel/carousel.ts [hunk 3]
```diff
@@ -39,8 +39,8 @@ export class NgbSlide {
...
+    '(mouseleave)': 'mouseEventCycle()',
```
Comment (benouat) [inline src/carousel/carousel.ts]: I would rename it to `onMouseLeave()`

File: src/carousel/carousel.ts [hunk 4]
```diff
@@ -84,6 +84,11 @@ export class NgbCarousel implements AfterContentChecked,
    */
   @Input() keyboard: boolean;
 
+  /**
+   * A flag to enable or disable mouse events
```
Comment (benouat) [inline src/carousel/carousel.ts]: `mouse events` is too generic in here. we only check for mouse enter/leave

I would rephrase it to `Flag to enable pause/resume on mouseover`.

**Comment by pkozlowski-opensource** [comment]
I did changes requested by @benouat, squashed and merged as 8d54cac.
